pandas/tests/reshape/test_from_dummies.py

```diff
@@ -0,0 +1,321 @@
+import numpy as np
+import pytest
+
+from pandas import (
+    DataFrame,
+    Series,
+)
+from pandas.core.reshape.reshape import from_dummies
+
+
+@pytest.fixture
+def dummies_basic():
+    return DataFrame(
+        {
+            "C": [1, 2, 3],
+            "col1_a": [1, 0, 1],
+            "col1_b": [0, 1, 0],
+            "col2_a": [0, 1, 0],
+            "col2_b": [1, 0, 0],
+            "col2_c": [0, 0, 1],
+        },
+    )
+
+
+@pytest.fixture
+def dummies_with_unassigned():
+    return DataFrame(
+        {
+            "C": [1, 2, 3],
+            "col1_a": [1, 0, 0],
+            "col1_b": [0, 1, 0],
+            "col2_a": [0, 1, 0],
+            "col2_b": [0, 0, 0],
+            "col2_c": [0, 0, 1],
+        },
+    )
+
+
+def test_from_dummies_to_series_basic():
+    dummies = DataFrame({"a": [1, 0, 0, 1], "b": [0, 1, 0, 0], "c": [0, 0, 1, 0]})
+    expected = Series(list("abca"))
+    result = from_dummies(dummies, to_series=True)
+    assert all(result == expected)
```

Comment:
use `tm.assert_frame_equal` and `tm.assert_series_equal` (see other tests for examples)